### PR TITLE
hellwal: init at 1.0.2

### PR DESCRIPTION
**Pull request by danihek**
Hello!

I wanted to add hellwal here, it's color palette generator, something like pywal.
Here's a link to repo: [hellwal](https://github.com/danihek/hellwal)

It's my first contribution here, so I don't really know If I did all correctly, please keep that in mind :)

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/applications/misc/hellwal/default.nix
Comment (pluiedev) [inline pkgs/applications/misc/hellwal/default.nix:1]: Please migrate to use the new by-name system (see pkgs/by-name/README.md)

File: pkgs/applications/misc/hellwal/default.nix
Comment (pluiedev) [inline pkgs/applications/misc/hellwal/default.nix:10]: Don't do this. Just get variables from pkgs via the lambda's parameters
```suggestion
```

File: pkgs/applications/misc/hellwal/default.nix
Comment (pluiedev) [inline pkgs/applications/misc/hellwal/default.nix:11]: Prefer `finalAttrs` over `rec` (see https://nixos.org/manual/nixpkgs/unstable/#mkderivation-recursive-attributes)
```suggestion
stdenv.mkDerivation (finalAttrs: {
```

File: pkgs/applications/misc/hellwal/default.nix
Comment (pluiedev) [inline pkgs/applications/misc/hellwal/default.nix:18]: Use `fetchFromGitHub` instead, and prefer `hash` over `sha256`
```suggestion
  src = fetchFromGitHub {
    owner = "danihek";
    repo = "hellwal";
    tag = "v${finalAttrs.version}";
    hash = "sha256-7/ldegQD/AbVeRfktNTb+Jem8DABBFks+VvntUyDvJA=";
  };
```

File: pkgs/applications/misc/hellwal/default.nix
Comment (pluiedev) [inline pkgs/applications/misc/hellwal/default.nix:30]:
```suggestion
    install -Dm755 hellwal -t $out/bin

    mkdir -p $out/share/docs/hellwal
    cp -r templates themes $out/share/docs/hellwal
```

File: pkgs/applications/misc/hellwal/default.nix
Comment (pluiedev) [inline pkgs/applications/misc/hellwal/default.nix:33]: Avoid with lib;
```suggestion
  meta = {
```

File: pkgs/applications/misc/hellwal/default.nix
Comment (pluiedev) [inline pkgs/applications/misc/hellwal/default.nix:35]: See pkgs/README.md for description wording guidelines
```suggestion
    description = "Fast, extensible color palette generator";
```

File: pkgs/applications/misc/hellwal/default.nix
Comment (pluiedev) [inline pkgs/applications/misc/hellwal/default.nix:40]: You also need to add a corresponding entry in `maintainers/maintainer-list.nix`
```suggestion
    maintainers = with lib.maintainers; [ danihek ];
```

**Review by pluiedev — CHANGES_REQUESTED** [review]
Welcome to Nixpkgs! There's a few issues that need to be resolved

Also, commit & PR should be titled `hellwal: init at 1.0.2`

File: pkgs/applications/misc/hellwal/default.nix
Comment (infinisil) [reply pkgs/applications/misc/hellwal/default.nix:1]: Weird, it should've automatically complained about that, there should never be a need to manually point this out. Looking into it

Comment (pluiedev) [reply pkgs/applications/misc/hellwal/default.nix:1]: Maybe it's because the CI is already erroring left and right and nixpkgs-vet just isn't run..?

**Comment by danihek** [comment]
How about now? Is it fine?

File: pkgs/applications/misc/hellwal/default.nix
Comment (infinisil) [reply pkgs/applications/misc/hellwal/default.nix]: It [did succeed](https://github.com/NixOS/nixpkgs/actions/runs/12285221762/job/34282751824?pr=364360) earlier, but I figured it out: Due to the use of `<nixpkgs>`, it failed to [evaluate with restricted eval](https://github.com/NixOS/nixpkgs-vet/blob/d6f01cfe8e1c7a0833d7840e4037aa95fab848dc/src/eval.rs#L198), in which case it [falls back](https://github.com/NixOS/nixpkgs-vet/blob/d6f01cfe8e1c7a0833d7840e4037aa95fab848dc/src/eval.rs#L466-L479) to assume such attributes can't be migrated. With the `<nixpkgs>` removed it would fail. Ping @NixOS/nixpkgs-vet for info :)

Comment (danihek) [reply pkgs/applications/misc/hellwal/default.nix]: @pluiedev sorry for bothering once again, could you point out what I'm doing wrong? 
It can't pass [nixpkgs-vet](https://github.com/NixOS/nixpkgs/actions/runs/12286137129/job/34285655226?pr=364360), 
and also fails here: 
https://github.com/NixOS/nixpkgs/actions/runs/12286137123/job/34285705423?pr=364360#step:5:576

Comment (infinisil) [reply pkgs/applications/misc/hellwal/default.nix]: The `nixpkgs-vet` error is admittedly a bit confusing, but please read it fully: https://github.com/NixOS/nixpkgs/actions/runs/12286137129/job/34285655226?pr=364360

Comment (infinisil) [reply pkgs/applications/misc/hellwal/default.nix]: To resolve the eval failure, you'll have to add yourself to [`maintainer-list.nix`](https://github.com/NixOS/nixpkgs/blob/master/maintainers/maintainer-list.nix) :)

Comment (danihek) [reply pkgs/applications/misc/hellwal/default.nix]: Thank you! I think I got it!

File: pkgs/top-level/all-packages.nix
Comment (pluiedev) [inline pkgs/top-level/all-packages.nix:3759]: Does it not work with clang on Darwin? Also, this is not necessary - just directly use gccStdenv within the derivation

File: pkgs/by-name/he/hellwal/package.nix
Comment (pluiedev) [inline pkgs/by-name/he/hellwal/package.nix:34]: Forgot to mention: add `meta.platforms`

**Review by pluiedev — CHANGES_REQUESTED** [review]

File: pkgs/top-level/all-packages.nix
Comment (danihek) [reply pkgs/top-level/all-packages.nix:3759]: I'm gonna be honest, it's like this only because it passed nixpkgs-vet - sorry I don't get all of things here yet.

File: pkgs/by-name/he/hellwal/package.nix
Comment (danihek) [reply pkgs/by-name/he/hellwal/package.nix:34]: added!

File: pkgs/top-level/all-packages.nix
Comment (danihek) [reply pkgs/top-level/all-packages.nix:3759]: So how should it look like?

Comment (infinisil) [reply pkgs/top-level/all-packages.nix:3759]: The error really could be improved, sorry for the confusion! :sweat_smile: 

I'll reword: `all-packages.nix` does not need to be changed at all in this PR, the directory `pkgs/by-name/he/hellwal` existing makes the attribute `pkgs.hellwal` exist too. So you can remove this entry here entirely :)

Comment (danihek) [reply pkgs/top-level/all-packages.nix]: Oh, it's fine, thank you for answering and teaching me :D
I did that, should be all good now :)

**Comment by danihek** [comment]
Thanks you all for helping me :D

I think it's ready right?

**Comment by pluiedev** [comment]
Needs to be squashed into two commits before merge at least :p 

* `maintainers: add danihek`
* `hellwal: init at 1.0.2`

**Comment by danihek** [comment]
Okay it's squashed :)

**Comment by danihek** [comment]
Okay, so:
- 2 commits
  - [maintainers: add danihek](https://github.com/NixOS/nixpkgs/pull/364360/commits/d886f603ab960164e16fe858796b0d91cf08b489)
  - [hellwal: init at 1.0.2](https://github.com/NixOS/nixpkgs/pull/364360/commits/713eac33bebb3c569df9ec0608e61a516a01c642)
- proper titles
- no errors checks

I think it's fine now. Thanks for patience and help, I learned a lot.

File: pkgs/by-name/he/hellwal/package.nix
Comment (pluiedev) [inline pkgs/by-name/he/hellwal/package.nix:32]: Why x86_64 specifically?

Comment (danihek) [reply pkgs/by-name/he/hellwal/package.nix:32]: Im pretty sure it failed previously before on aarch64, both linux and darwin

Comment (pluiedev) [reply pkgs/by-name/he/hellwal/package.nix]: I'm not sure about `platforms.all` either. Set it to `platforms.unix` for now.

Comment (danihek) [reply pkgs/by-name/he/hellwal/package.nix]: Done: [hellwal: init at 1.0.2](https://github.com/NixOS/nixpkgs/pull/364360/commits/64b5ea8e481a379b337ee0959f7e74d6c3e5d6e2)

**Comment by pluiedev** [comment]
@ofborg build hellwal

**Comment by philiptaron** [comment]
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 364360`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>hellwal</li>
  </ul>
</details>

**Review by philiptaron — APPROVED** [review]

**Comment by danihek** [comment]
@pluiedev Thank you very much for helping me :)

**Pull request merged**
